docs/jsonapi-spec.textile

```diff
@@ -612,7 +673,36 @@ Fields included in the filter clause in addition to the @_id@ are ignored.
+# *Concurrency Faults*: These occur when the a document is under very high concurrenct modification which make it impossible for the JSON API to complete an optimistic "compare-and-set" update. Modification operations read documents from the storage platform and update or delete it only if it has not changed since being read. If the document has changed, the filter that seleted the document is tested again on that document, and the modification is re-tried. Under high contention it is possible for one request to "starve" and be unable to complete.     
+
+In the context of Multi Document Failures we consider a failure (after a configured number of retries) from either of these categories as a failure to complete a command.  
+
+*Note* Neither of these faults will leave a document in an inconsistent state, that would cause "eventually consistent" reads. Document retain internal consistency as they are stored on a single row which is always atomically updated. And they maintain cluster wide consistency through the user of cluster wide "compare-and-set" operations.    
+
+h4(#considerationsMultiDocumentFailureFailFast). Fail Fast Multi Document Failure
+
+Under the "fail fast" mode each operation on the database is performed sequentially, ensuring the previous operation has succedded before starting the next. In this mode a single failure causes the command to stop processing and return information on the work that has completed together with an error. 
+
+For example, consider an ordered "@insertMany@":#commandInsertMany command that is inserting three documents. This command will use "fail fast", the second document will be inserted once the first is successful. If inserting the second document fails the command will return both the @_id@ of the first document and an error describing the failure inserting the second. No attempt will be made to insert the third document. 
+
+h4(#considerationsMultiDocumentFailureFailSilently). Fail Silently Multi Document Failure
+
+Under the "fail silently" mode the command will attempt to perform all operations on the database, which may include successfully processing some operations after others have failed. In this mode a single failure does not cause the command to stop processing, instead it makes a best effort to complete and returns information on the work that was completed and potentially what was not. 
+
+Fail silently also allows the server to re-order the database operations to achieve best performance, which may include running multiple operations in parallel. 
+
+Commands that use Fail Silently may return an error and status information about what was successfully completed. See "Response Messages":#messagingResponse for the structure of responses, and "Commands":#commands for individual command responses. 
```

Comment:
We need a definition here: 
* what status to return in case of a partial failure
* do we return all errors we hit or just some (limit)